File: strings/compareVersionNumbers.java
// https://leetcode.com/problems/compare-version-numbers/
package strings;

public class compareVersionNumbers {
    public static void main(String[] args) {
        SolutionCVN solution = new SolutionCVN();
        System.out.println(solution.compareVersion("1.0.1", "1.0"));
        System.out.println(solution.compareVersion("1.0.1", "1.0.0"));
        System.out.println(solution.compareVersion("7.5.2.4", "7.5.3"));
    }
}


class SolutionCVN {

    // Submitted by @Jiganesh

    // TC : O(N)
    // SC : O(N)

    // Runtime: 1 ms, faster than 90.83% of Java online submissions for Compare Version Numbers.
    // Memory Usage: 42.5 MB, less than 8.43% of Java online submissions for Compare Version Numbers.
    
    public int compareVersion(String version1, String version2) {
        
        String[] v1 = version1.split("\\.");
        String[] v2 = version2.split("\\.");
        
        for (int i =0 ; i< Math.max(v1.length, v2.length); i++){

            int val1 = (i<v1.length)?Integer.parseInt(v1[i]):0;
            int val2 = (i<v2.length)?Integer.parseInt(v2[i]):0;
            
            if (val1 < val2) return -1;
            else if(val1 > val2) return 1; 
        }
        return 0;
    }
}

File: strings/compareVersionNumbers.py
# https://leetcode.com/problems/compare-version-numbers/
class Solution(object):
    
    # Submitted by @Jiganesh
     
    # TC : O(N)
    # SC : O(N)
    
    # Runtime: 22 ms, faster than 58.55% of Python online submissions for Compare Version Numbers.
    # Memory Usage: 13.6 MB, less than 21.71% of Python online submissions for Compare Version Numbers.
    
    def compareVersion(self, version1, version2):
        """
        :type version1: str
        :type version2: str
        :rtype: int
        """
        
        a = version1.split(".")
        b = version2.split(".")
        
        for i in range(max(len(a), len(b))):
            
            vala = i< len(a) and int(a[i]) or 0
            valb = i< len(b) and int(b[i]) or 0
            
            if vala<valb : return -1
            elif vala>valb : return 1
            
        return 0